controllers/apps/apimanager_managed_routes_handler.go

```diff
@@ -0,0 +1,107 @@
+package controllers
+
+import (
+	"context"
+
+	appscommon "github.com/3scale/3scale-operator/apis/apps"
+	appsv1alpha1 "github.com/3scale/3scale-operator/apis/apps/v1alpha1"
+	"github.com/3scale/3scale-operator/pkg/3scale/amp/component"
+	"github.com/go-logr/logr"
+	appsv1 "github.com/openshift/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ handler.Mapper = &APIManagerRoutesHandler{}
+
+// APIManagerRoutesHandler is an EventHandler that maps an existing OpenShift
+// route to an APIManager. This handler should only be used on Route objects
+// and when APIManager is used.
+type APIManagerRoutesHandler struct {
+	k8sclient client.Client
+	logger    logr.Logger
+}
+
+func (h *APIManagerRoutesHandler) Map(mapObject handler.MapObject) []reconcile.Request {
+	var res []reconcile.Request
+	apimanagerReconcileRequest := h.getAPIManagerOwnerReconcileRequest(mapObject.Meta)
```

Comment:
At the moment I prefer to have it separate, also because of the recursion so I prefer to be done in its own context. Both approaches seem ok to me but I'll go for the separate one